src/de/jost_net/JVerein/gui/action/BuchungGegenbuchungAction.java

```diff
@@ -0,0 +1,85 @@
+/**********************************************************************
+ * Copyright (c) by Heiner Jostkleigrewe
+ * This program is free software: you can redistribute it and/or modify it under the terms of the 
+ * GNU General Public License as published by the Free Software Foundation, either version 3 of the 
+ * License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,  but WITHOUT ANY WARRANTY; without 
+ *  even the implied warranty of  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See 
+ *  the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program.  If not, 
+ * see <http://www.gnu.org/licenses/>.
+ * 
+ * heiner@jverein.de
+ * www.jverein.de
+ **********************************************************************/
+package de.jost_net.JVerein.gui.action;
+
+import de.jost_net.JVerein.Einstellungen;
+import de.jost_net.JVerein.gui.dialogs.KontoAuswahlDialog;
+import de.jost_net.JVerein.gui.view.BuchungView;
+import de.jost_net.JVerein.rmi.Buchung;
+import de.jost_net.JVerein.rmi.Konto;
+import de.willuhn.datasource.rmi.DBIterator;
+import de.willuhn.datasource.rmi.DBService;
+import de.willuhn.jameica.gui.Action;
+import de.willuhn.jameica.gui.GUI;
+import de.willuhn.jameica.system.OperationCanceledException;
+import de.willuhn.util.ApplicationException;
+
+public class BuchungGegenbuchungAction implements Action
+{
+  
+  @Override
+  public void handleAction(Object context) throws ApplicationException
+  {
+ 
+    if (context == null || !(context instanceof Buchung))
+    {
+      throw new ApplicationException("Keine Buchung ausgew�hlt");
+    }
+    Buchung b = (Buchung) context;
+    Konto konto = null;
+    try
+    {
+      final DBService service = Einstellungen.getDBService();
+      DBIterator<Konto> konten = service.createList(Konto.class);
+      konten.addFilter("buchungsart = ?", b.getBuchungsartId());
+      if (konten.size() > 0)
+      {
+        konto = (Konto) konten.next();
+      }
+      if (konto == null)
+      {
+        KontoAuswahlDialog d = new KontoAuswahlDialog(
+            KontoAuswahlDialog.POSITION_CENTER, false, false, true);
+        konto = (Konto) d.open();
+      }
+      if (konto != null)
+      {
+        Buchung bu = (Buchung) Einstellungen.getDBService().createObject(Buchung.class,
+            null);
+        bu.setKonto(konto);
+        bu.setName(b.getName());
+        bu.setBetrag(-b.getBetrag());
+        bu.setZweck(b.getZweck());
+        bu.setDatum(b.getDatum());
+        if (b.getBuchungsart() != null)
+          bu.setBuchungsart(b.getBuchungsartId());
+        if (b.getProjekt() != null)
```

Comment:
Führt das Setzen des Projekts nicht dazu, dass sich die beiden Buchungen im Projektsaldo annullieren? Mir fällt gerade kein Szenario ein, in dem man das will.

Reply:
Also bei mir heben sich Umbuchungen auch im normalen Buchungsklassen Saldo gegenseitig auf. Dafür sind es ja Umbuchungen und keine Einnahmen oder Ausgaben. Und ich erlaube ja die Aktion nur bei Umbuchungen. 
Z. B. Eine Umbuchung von Geld vom Girokonto auf ein Anlagekonto muss sich aufheben weil das weder Einnahmen noch Ausgaben sind. Der Geldbestand bleibt ja gleich. 
Oder wenn ich eine Einzahlung für eine Kaution habe ist das auch keine Einnahme sondern ich mache eine Gegenbuchung im Kautionskonto. Ich habe mehr Geld auf dem Girokonto aber gleichzeitig mehr Schulden.
Wenn die Buchungen jetzt einem Projekt zugeordnet sind sollte das doch innerhalb des Projektes genauso sein nehme ich mal an.

Reply:
> Also bei mir heben sich Umbuchungen auch im normalen Buchungsklassen Saldo gegenseitig auf. Dafür sind es ja Umbuchungen und keine Einnahmen oder Ausgaben. Und ich erlaube ja die Aktion nur bei Umbuchungen. Z. B. Eine Umbuchung von Geld vom Girokonto auf ein Anlagekonto muss sich aufheben weil das weder Einnahmen noch Ausgaben sind. Der Geldbestand bleibt ja gleich. Oder wenn ich eine Einzahlung für eine Kaution habe ist das auch keine Einnahme sondern ich mache eine Gegenbuchung im Kautionskonto. Ich habe mehr Geld auf dem Girokonto aber gleichzeitig mehr Schulden.
Wenn die Buchungen jetzt einem Projekt zugeordnet sind sollte das doch innerhalb des Projektes genauso sein nehme ich mal an.

Hier stimme ich @JohannMaierhofer zu.

Reply:
Warum sollte die Umbuchung dann überhaupt einem Projekt zugeordnet werden?

Reply:
Ich denke, das hängt dann davon ab ob man auch an anderen Stellen die Buchungen als Teil des Projekts sehen möchte z.B. in den Buchungslisten.
Für das Feature ist das aber egal. 
Entweder ist eine Buchung nicht Teil eines Projektes und dann wird auch nichts mit kopiert.
Oder es ist Teil eines Projektes und dann wird es halt mit kopiert. Sollte dann aus irgendeinem Grund das Projekt nicht in der Gegenbuchung gewünscht sein, dann kann man es ja vor dem abspeichern immer noch rauslöschen. Aber das Behalten dürfte in dem Fall wohl der Standard sein.

Reply:
Ich ordne Umbuchungen auch immer einem Projekt zu. Es ist aber wie schon richtig beschrieben jedem selbst überlassen ob er das macht oder nicht.